### PR TITLE
Fix Editorial Nits

### PR DESCRIPTION
**Pull request by RexJaeschke**
@BillWagner, as I was working on the V8 spec for patterns, I had need to review the newly added chapter 7 (patterns), the first time I have done so since we agreed to break-off that text into its own chapter. Here are the editorial changes I made:

- Made section headings sentence-case rather than title-case.
- 11.2 contained text *prior* to its first child section, 11.2.1, which we don't allow, so I added a new section 11.2.1 General.
- I removed italics from references to terms that have already been defined.
- I added a bunch of xrefs, especially those placeholders flagged using `(XREF)`.
- Removed all occurrences of the the first-person pronoun "we" in newer text by rewriting.

### CODE REVIEW

File: standard/expressions.md [hunk 1]
```diff
@@ -4349,12 +4349,12 @@ User defined conversions are not considered by the `is` operator.
...
 #### 12.12.12.2 The is-pattern operator
 
-The *is-pattern operator* is used to check if the value computed by an expression *matches* a given pattern (XREF TO DEF OF “PATTERN MATCHES”). The check is performed at runtime. The result of the is-pattern operator is true if the value matches the pattern; otherwise it is false.
+The *is-pattern operator* is used to check if the value computed by an expression *matches* a given pattern ([§11](patterns.md#11-patterns-and-pattern-matching)). The check is performed at runtime. The result of the is-pattern operator is true if the value matches the pattern; otherwise it is false.
 
 For an expression of the form `E is P`, where `E` is a relational expression of type `T` and `P` is a pattern, it is a compile-time error if any of the following hold:
 
 - `E` does not designate a value or does not have a type.
-- The pattern `P` is not applicable (XREF NEEDED) to the type `T`.
+- The pattern `P` is not applicable to the type `T`.
```
Comment (jskeet) [inline standard/expressions.md]: Link to 11.2?

File: standard/patterns.md [hunk 2]
```diff
@@ -18,7 +20,7 @@ pattern
 
 A *declaration_pattern* and a *var_pattern* can result in the declaration of a local variable.
 
-Each pattern form defines the set of types for input values that the pattern may be applied to.  We say a pattern `P` is *applicable to* a type `T` if `T` is among the types whose values the pattern may match.  It is an error if a pattern `P` appears in a program to match a *pattern input value* of type `T` if `P` is not applicable to `T`.
+Each pattern form defines the set of types for input values that the pattern may be applied to.  A pattern `P` is *applicable to* a type `T` if `T` is among the types whose values the pattern may match.  It is an error if a pattern `P` appears in a program to match a pattern input value ([§11.1](patterns.md#111-general)) of type `T` if `P` is not applicable to `T`.
```
Comment (jskeet) [inline standard/patterns.md]: Remove double space after the period at the same time?

File: standard/patterns.md [hunk 3]
```diff
@@ -129,28 +131,28 @@ designation
...
 It is an error if the name `var` would bind to a type where a *var_pattern* is used.
 
-## 11.3 Pattern Subsumption
+## 11.3 Pattern subsumption
 
-In a switch statement, it is an error if a case’s pattern is *subsumed* by the preceding set of unguarded cases (XREF).
+In a switch statement, it is an error if a case’s pattern is *subsumed* by the preceding set of unguarded cases ([§13.8.3](statements.md#1383-the-switch-statement)).
 Informally, this means that any input value would have been matched by one of the previous cases.
-Here we define when a set of patterns *subsumes* a given pattern.
+Here is defined when a set of patterns *subsumes* a given pattern.
```
Comment (jskeet) [inline standard/patterns.md]: I'm not keen on this sentence at the moment. How about "The following rules define when a set of patterns *subsumes* a given pattern:"

File: standard/patterns.md [hunk 4]
```diff
@@ -129,28 +131,28 @@ designation
...
+- `P` is a var pattern and the set of patterns `Q` is *exhaustive* ([§11.4](patterns.md#114-pattern-exhaustiveness)) for the type of the pattern input value ([§11.1](patterns.md#111-general)), and either the pattern input value is not of a nullable type or some pattern in `Q` would match `null`.
+- `P` is a declaration pattern with type `T` and the set of patterns `Q` is *exhaustive* for the type `T` ([§11.4](patterns.md#114-pattern-exhaustiveness)).
 
-## 11.4 Pattern Exhaustiveness
+## 11.4 Pattern exhaustiveness
 
-Informally, we say that a set of patterns is exhaustive for a type if some pattern in the set is applicable to every possible value of that type other than null.
-Here we define when a set of patterns is *exhaustive* for a type.
+Informally, a set of patterns is exhaustive for a type if some pattern in the set is applicable to every possible value of that type other than null.
+Here is defined when a set of patterns is *exhaustive* for a type.
```
Comment (jskeet) [inline standard/patterns.md]: Same sort of change here.

File: standard/statements.md [hunk 5]
```diff
@@ -875,12 +875,12 @@ The *statement_list*s of a *switch_block* may contain declaration statements ([
 A switch label is reachable if at least one of the following is true:
 
 - The switch expression is a constant value and either
-  - the label is a `case` whose pattern *would match* (XREF to “would match” in patterns.md) that value, and label’s guard is either absent or not a constant expression with the value false; or
+  - the label is a `case` whose pattern *would match* that value, and label’s guard is either absent or not a constant expression with the value false; or
```
Comment (jskeet) [inline standard/statements.md]: Xref to 11.4?

File: standard/statements.md [hunk 6]
```diff
@@ -890,8 +890,8 @@ The end point of a `switch` statement is reachable if the switch statement is re
 
 - The `switch` statement contains a reachable `break` statement that exits the `switch` statement.
 - No `default` label is present and either
-  - The switch expression is a non-constant value, and the set of patterns appearing among the cases of the switch statement that do not have guards or have guards whose value is the constant true, is not *exhaustive* (NEED XREF) for the switch governing type.
-  - The switch expression is a non-constant value of a nullable type, and no pattern appearing among the cases of the switch statement that do not have guards or have guards whose value is the constant true would match the value `null` (XREF to “would match” in patterns).
+  - The switch expression is a non-constant value, and the set of patterns appearing among the cases of the switch statement that do not have guards or have guards whose value is the constant true, is not *exhaustive* for the switch governing type.
```
Comment (jskeet) [inline standard/statements.md]: Xref to 11.4 again? I can see we could have too many of them...

File: standard/types.md [hunk 7]
```diff
@@ -621,7 +621,7 @@ As a type, type parameters are purely a compile-time construct. At run-time, eac
 
 ## 8.6 Expression tree types
 
-***Expression trees*** permit lambda expressions to be represented as data structures instead of executable code. Expression trees are values of ***expression tree types*** of the form `System.Linq.Expressions.Expression<TDelegate>`, where `TDelegate` is any delegate type. For the remainder of this specification we will refer to these types using the shorthand `Expression<TDelegate>`.
+***Expression trees*** permit lambda expressions to be represented as data structures instead of executable code. Expression trees are values of ***expression tree types*** of the form `System.Linq.Expressions.Expression<TDelegate>`, where `TDelegate` is any delegate type. For the remainder of this specification these types will be referred to using the shorthand `Expression<TDelegate>`.
```
Comment (jskeet) [inline standard/types.md:624]: Standard instead of specification?

**Review by jskeet — APPROVED** [review]
A few nits (some of which I'm not sure about) but generally fine. Let's try to get this in before next week's meeting.

File: standard/expressions.md (hunk 1 above)
Comment (BillWagner) [reply standard/expressions.md]: Adding a suggestion:

```suggestion
- The pattern `P` is not applicable (§11.2) to the type `T`.
```

File: standard/patterns.md (hunk 2 above)
Comment (BillWagner) [reply standard/patterns.md]: adding a suggestion:

```suggestion
Each pattern form defines the set of types for input values that the pattern may be applied to. A pattern `P` is *applicable to* a type `T` if `T` is among the types whose values the pattern may match. It is an error if a pattern `P` appears in a program to match a pattern input value ([§11.1](patterns.md#111-general)) of type `T` if `P` is not applicable to `T`.
```

File: standard/patterns.md [hunk 8]
```diff
@@ -40,9 +42,9 @@ single_variable_designation
...
+Certain combinations of static type of the pattern input value and the given type are considered incompatible and result in a compile-time error. A value of static type `E` is said to be ***pattern compatible*** with the type `T` if there exists an identity conversion, an implicit reference conversion, a boxing conversion, an explicit reference conversion, or an unboxing conversion from `E` to `T`, or if either `E` or `T` is an open type ([§8.4.3](types.md#843-open-and-closed-types)).  A declaration pattern naming a type `T` is *applicable to* every type `E` for which `E` is pattern compatible with `T`.
```
Comment (BillWagner) [inline standard/patterns.md]: remove double space after period:

```suggestion
Certain combinations of static type of the pattern input value and the given type are considered incompatible and result in a compile-time error. A value of static type `E` is said to be ***pattern compatible*** with the type `T` if there exists an identity conversion, an implicit reference conversion, a boxing conversion, an explicit reference conversion, or an unboxing conversion from `E` to `T`, or if either `E` or `T` is an open type ([§8.4.3](types.md#843-open-and-closed-types)). A declaration pattern naming a type `T` is *applicable to* every type `E` for which `E` is pattern compatible with `T`.
```

File: standard/patterns.md [hunk 9]
```diff
@@ -129,28 +131,28 @@ designation
...
+Given a pattern input value ([§11.1](patterns.md#111-general)) *e*, if *designation* is the *identifier* `_`, it denotes a discard ([§9.2.9.1](variables.md#9291-discards)), and the value of *e* is not bound to anything. (Although a declared variable with that name may be in scope at that point, that named variable is not seen in this context.) If *designation* is any other identifier, at runtime the value of *e* is bound to a newly introduced local variable ([§9.2.9](variables.md#929-local-variables)) of that name whose type is the static type of *e*, and the pattern input value is assigned to that local variable.
```
Comment (BillWagner) [inline standard/patterns.md:134]: This is the closest line where I can comment, and I can't make a suggestion:

Line 119 (A *var_pattern* matches every value.  That is, a pattern-matching operation with a *var_pattern* always succeeds.) has a double space at the end of the first sentence.

File: standard/patterns.md (hunk 3 above)
Comment (BillWagner) [reply standard/patterns.md]: converting to a suggestion:

```suggestion
The following rules define when a set of patterns subsumes a given pattern:
```

File: standard/patterns.md (hunk 4 above)
Comment (BillWagner) [reply standard/patterns.md]: creating suggestion:

```suggestion
The following rules define when a set of patterns is *exhaustive* for a type:
```

File: standard/statements.md (hunk 5 above)
Comment (BillWagner) [reply standard/statements.md]: I was wondering about the new pattern forms general clause, where "match" is defined.

File: standard/statements.md (hunk 6 above)
Comment (BillWagner) [reply standard/statements.md]: This should be 11.4. Adding suggestion:

```suggestion
  - The switch expression is a non-constant value, and the set of patterns appearing among the cases of the switch statement that do not have guards or have guards whose value is the constant true, is not *exhaustive* (§11.4) for the switch governing type.
```

File: standard/variables.md [hunk 10]
```diff
@@ -330,7 +330,7 @@ The definite-assignment state of *v* at the beginning of *expr* is the same as t
 The definite-assignment state of *v* at the beginning of a case’s guard clause is
 
 - If *v* is a pattern variable declared in the *switch_label*: “definitely assigned”.
-- If the switch label containing that guard clause is not reachable (XREF): “definitely assigned”.
+- If the switch label containing that guard clause is not reachable: “definitely assigned”.
```
Comment (BillWagner) [inline standard/variables.md]: Should this link to 13.8.3?

```suggestion
- If the switch label containing that guard clause (§13.8.3) is not reachable: “definitely assigned”.
```

**Review by BillWagner — APPROVED** [review]
I saw a few nits here as well. Also, converted some of Jon's comments to suggestions.

Overall, this LGTM.

File: standard/statements.md [hunk 11]
```diff
@@ -676,9 +676,9 @@ The ***governing type*** of a `switch` statement is established by the switch ex
 
 There can be at most one `default` label in a `switch` statement.
 
-It is an error if the pattern of any switch label is not *applicable* (NEED AN XREF) to the type of the input expression.
+It is an error if the pattern of any switch label is not *applicable* (§pattern-forms-general) to the type of the input expression.
 
-It is an error if the pattern of any switch label is *subsumed* by (NEED AN XREF) the set of patterns of earlier switch labels of the switch statement that do not have a case guard or whose case guard is a constant expression with the value true.
+It is an error if the pattern of any switch label is *subsumed* by ([§11.3]patterns.md#113-pattern-subsumption) the set of patterns of earlier switch labels of the switch statement that do not have a case guard or whose case guard is a constant expression with the value true.
```
Comment (BillWagner) [inline standard/statements.md]: This is causing the error in the renumbering tool. The link is malformed. I just removed the MD link syntax so the tool will add it when it runs:

```suggestion
It is an error if the pattern of any switch label is *subsumed* by (§11.3) the set of patterns of earlier switch labels of the switch statement that do not have a case guard or whose case guard is a constant expression with the value true.
```

File: standard/statements.md [hunk 12]
```diff
@@ -875,12 +875,12 @@ The *statement_list*s of a *switch_block* may contain declaration statements ([
...
+  - the label is a `case` whose pattern *would match* that value, and label’s guard is either absent or not a constant expression with the value false; or
   - it is a `default` label, and no switch section contains a case label whose pattern would match that value, and whose guard is either absent or a constant expression with the value true.
 - The switch expression is not a constant value and either
   - the label is a `case` without a guard or with a guard whose value is not the constant false; or
   - it is a `default` label and
-    - the set of patterns appearing among the cases of the switch statement that do not have guards or have guards whose value is the constant true, is not *exhaustive* (NEED XREF) for the switch controlling type; or
+    - the set of patterns appearing among the cases of the switch statement that do not have guards or have guards whose value is the constant true, is not *exhaustive* ([§11.4]patterns.md#114-pattern-exhaustiveness) for the switch controlling type; or
```
Comment (BillWagner) [inline standard/statements.md]: Same issue with link format:

```suggestion
    - the set of patterns appearing among the cases of the switch statement that do not have guards or have guards whose value is the constant true, is not *exhaustive* (§11.4) for the switch controlling type; or
```

File: standard/types.md (hunk 7 above)
Comment (RexJaeschke) [reply standard/types.md:624]: Actually, "specification" is the preferred ISO term, and we're not consistent. I'll change the (few) occurrences of "Standard" to "specification."

**Pull request merged**